resources/function_help/json/regexp_matches

```diff
@@ -0,0 +1,11 @@
+{
+  "name": "regexp_matches",
+  "type": "function",
+  "description": "Returns an array of all strings captured by capturing groups, in the order the groups themselves appear in the supplied regular expression against a string.<br>If matches are found, the first element of the returned array will contain the text that mathed the full regular expression pattern.",
+  "arguments": [
+    {"arg":"string", "description":"the string to capture groups from against the regular expression"},
+    {"arg":"regex","description":"the regular expression used to capture groups"},
+    {"arg":"emptyvalue","optional":true,"default":"''","description":"the optional string to use as replacement to empty array values"}],
+  "examples": [ { "expression":"regexp_matches('qgis=>rocks;hello=>world','qgis=>(.*)[;$]')", "returns":"array: 'qgis=>rocks;', 'rocks'"}
```

Comment:
The empty_value is useful when you use regexp_matches to build a label and want to display a default string when the matching group returns an empty string. 

It prevents this rather ugly scenario: `if(array_get(regexp_matches(...),0) != '', regexp_matches(...), "empty value string")`